bump_definitions.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Axis Communications AB.
+# For a full list of individual contributors, please see the commit history.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Creates new versions of existing type definitions by increasing the
+desired version position (major, minor, or patch) and creating a new
+matching file.
+"""
+
+import argparse
+import fnmatch
+import re
+import sys
+import time
+from pathlib import Path
+from typing import List
+
+import semver
+
+import versions
+
+
+def _bump_versions(root: Path, position: str, pattern: str) -> List[Path]:
+    """Find definition files in the given directory, match their types
+    against the given glob pnattern, and create new files with version
+    bumps in the selected position. Returns the paths of the created
+    files.
+    """
+    result = []
+    for typename, version in versions.latest_in_dir(root).items():
+        if not fnmatch.fnmatch(typename, pattern):
+            continue
+
+        old_path = root / typename / f"{version}.yml"
+        new_version = getattr(version, "bump_" + position)()
+        new_path = old_path.with_name(f"{new_version}.yml")
+        new_path.write_text(
+            _transform_definition(old_path.read_text(encoding="utf-8"), new_version),
+            encoding="utf-8",
+        )
+        result.append(new_path)
+    return result
+
+
+def _transform_definition(
+    old_definition: str, version: semver.version.Version, copyright_year: int = None
```

Comment:
The parameter copyright_year is not provided when calling this method. Is that intentional and left for future improvement?

Reply:
This optional parameter is only used for tests. If omitted, the current year will be used (see line 68). We could of course make the argument mandatory and move `time.localtime(time.time()).tm_year` to the callsite on line 51. I have no strong opinion in the matter.

Reply:
I don't see that we currently will have the need to set the year to something else than current year, so let's keep it out of the CLI for now at least.